proto/schema.proto

```diff
@@ -0,0 +1,201 @@
+syntax="proto3";
+
+option go_package = "github.com/kubernetes-csi/external-snapshot-metadata/pkg/grpc";
+
+// The proposed Kubernetes SnapshotMetadata API is very similar to the CSI
+// SnapshotMetadata Service API. The only structural differences between the
+// two specifications are in some message properties.
+// Refs: https://github.com/kubernetes/enhancements/blob/ffb78131af83279b8852ae742ca60ff81498eecd/keps/sig-storage/3314-csi-changed-block-tracking/README.md#the-csi-snapshotmetadata-service-api
+// https://github.com/container-storage-interface/spec/pull/551
+service SnapshotMetadata {
+  rpc GetMetadataAllocated(GetMetadataAllocatedRequest)
+    returns (stream GetMetadataAllocatedResponse) {}
+
+  rpc GetMetadataDelta(GetMetadataDeltaRequest)
+    returns (stream GetMetadataDeltaResponse) {}
+}
+
+enum BlockMetadataType {
+  UNKNOWN=0;
+
+  // The FIXED_LENGTH value indicates that data ranges are
+  // returned in fixed size blocks.
+  FIXED_LENGTH=1;
+
+  // The VARIABLE_LENGTH value indicates that data ranges
+  // are returned in potentially variable sized extents.
+  VARIABLE_LENGTH=2;
+}
+
+// BlockMetadata specifies a data range.
+message BlockMetadata {
+  // This is the zero based byte position in the volume or snapshot,
+  // measured from the start of the object.
+  // This field is REQUIRED.
+  int64 byte_offset = 1;
+
+  // This is the size of the data range.
+  // size_bytes will be greater than zero.
+  // This field is REQUIRED.
+  int64 size_bytes = 2;
+}
+
+// The GetMetadataAllocatedRequest message is used to solicit metadata
+// on the allocated blocks of a snapshot: i.e. this identifies the
+// data ranges that have valid data as they were the target of some
+// previous write operation on the volume.
+message GetMetadataAllocatedRequest {
+  // This field specifies an audience scoped authentication token presented by client.
+  // This token is used to authenticate and authorize the request at server side.
+  // This field is REQUIRED.
+  string security_token = 1;
+
+  // This field indicates the Namespace where the VolumeSnapshot exists.
+  // If not set, the VolumeSnapshot will be looked for in the "default" Namespace.
+  // This field is OPTIONAL.
+  string namespace = 2;
```

Comment:
I don't think this should be optional. If the client can look up the `VolumeSnapshot` object they can easily copy out the namespace value!